### PR TITLE
UI datimp 961

### PR DESCRIPTION
**Pull request by Mathansivavelmurugesan**
UIDATIMP-961 -  Cover <WithValidation> component with tests
## Purpose
Covered the module with unit test cases and obtained 100% coverage i code coverage.

## Approach
Rendered the component by mocking the validation and children props passed to the component.

### CODE REVIEW

File: src/components/WithValidation/WithValidation.test.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  );
+
+  return renderWithContext(component);
+};
+
+describe('With Validation component', () => {
+  it('Render the component', () => {
+    waitFor(() => expect(renderWithValidation(false)).toBeDefined());
+  });
+  it('Check the validateQuotedStringOrMarcPath by passing normal string and setting remove value prop as true', () => {
```
Comment (OleksandrHladchenko1) [inline src/components/WithValidation/WithValidation.test.js:65]: Separate long description by adding **describe**

File: src/components/WithValidation/WithValidation.test.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    const { getByTestId } = renderWithValidation(true);
+    const input = getByTestId('name');
+
+    waitFor(() => expect(input.value).toBe('Hello'));
+  });
+  it('Check the validateQuotedStringOrMarcPath by passing null and setting remove as false', () => {
```
Comment (OleksandrHladchenko1) [inline src/components/WithValidation/WithValidation.test.js:71]: Here the same

File: src/components/WithValidation/WithValidation.test.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import '../../../test/jest/__mock__';
+
+import { WithValidation } from './WithValidation';
+
+const TestComponent = props => {
+  let data = '';
+  const handleChange = () => {
+    const value = props.validate();
+
+    data = value;
```
Comment (OleksandrHladchenko1) [inline src/components/WithValidation/WithValidation.test.js:16]: It is no use to declare **value** variable

File: src/components/WithValidation/WithValidation.test.js (hunk 1 above)
Comment (Mathansivavelmurugesan) [reply src/components/WithValidation/WithValidation.test.js]: Added describe and seperated description.

File: src/components/WithValidation/WithValidation.test.js (hunk 2 above)
Comment (Mathansivavelmurugesan) [reply src/components/WithValidation/WithValidation.test.js]: Added describe and seperated description.

File: src/components/WithValidation/WithValidation.test.js (hunk 3 above)
Comment (Mathansivavelmurugesan) [reply src/components/WithValidation/WithValidation.test.js]: Removed that variable.

File: src/components/WithValidation/WithValidation.test.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+
```
Comment (mariia-aloshyna) [inline src/components/WithValidation/WithValidation.test.js:2]: Remove the empty line, please
See code style for imports sequence here - https://wiki.folio.org/pages/resumedraft.action?draftId=14458784&draftShareId=483fa2fb-615e-4002-a916-c976bd5fb8c6&

File: src/components/WithValidation/WithValidation.test.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      {children}
+    </WithValidation>
+  );
+
+  return renderWithContext(component);
+};
+
+describe('With Validation component', () => {
+  it('Render the component', () => {
+    waitFor(() => expect(renderWithValidation(false)).toBeDefined());
```
Comment (mariia-aloshyna) [inline src/components/WithValidation/WithValidation.test.js:61]: Do you actually need `waitFor` here?

**Comment by Mathansivavelmurugesan** [comment]
Changing branch name

**Pull request closed without merging**